src/main/scala/mesosphere/marathon/raml/AppConversion.scala

```diff
@@ -137,6 +138,11 @@ trait AppConversion extends DefaultConversions with CheckConversion with Constra
 
     val versionInfo = state.VersionInfo.OnlyVersion(app.version.map(Timestamp(_)).getOrElse(Timestamp.now()))
 
+    val role = app.role.orNull
+    if (role == null) {
+      throw new IllegalArgumentException("Failed to deserialize raml.App, no role provided")
+    }
```

Comment:
This should be more Scala like
val role = app.role.getOrElse(throw new IllegalArgumentException("Failed to convert raml.App, no role provided. This is a bug in AppNormalization."))

Reply:
Ah, right. Changed.